Remove a dead link and in the "User enumeration" section, update variable names. (#110)
cfrg/draft-irtf-cfrg-opaque: draft-irtf-cfrg-opaque.md

User enumeration refers to attacks where the attacker tries to learn
whether a given user identity is registered with a server. Preventing
such attacks requires the server to act with unknown user identities
in a way that is indistinguishable from its behavior with existing
users. Here we suggest a way to implement such defense, namely, a way for
simulating the values Z and envU for non-existing users.
Note that if the same pair of user identity idU and value M is received
twice by the server, the response needs to be the same in both cases (since
this would be the case for real users).
For protection against this attack, one would apply the encryption function in

> the construction of envU to all the key material in envU, namely,

Reviewer: ```suggestion
the construction of envU to all the key material in envU.
```

> cleartext_creds will be empty.

Reviewer: ```suggestion
```

The server S will have two keys MK, MK' for a PRF f
(this refers to a regular PRF such as HMAC or CMAC).

> Upon receiving a pair of user identity idU and value M for a non-existing

Reviewer: ```suggestion
Upon receiving a CredentialRequest for a non-existing
```

user idU, S computes kU=f(MK; idU) and kU'=f(MK'; idU) and responds with

> values Z=M^kU and envU, where the latter is computed as follows.
> rwdU is set to kU' and secret_creds is set to the all-zero string (of the

Reviewer: ```suggestion
CredentialResponse carrying Z=M^kU and envU, where the latter is computed as follows.
rwdU is set to kU' and secret_creds is set to the all-zero string (of the
```

There is one form of leakage that the above allows and whose prevention would
require a change in OPAQUE.

> <span class="x x-first x-last">An </span>attacker that <span class="x x-first x-last">attempts authentication with the</span> same <span class="x x-first x-last">CredentialRequest</span> twice and receives

Reviewer: ```suggestion
An attacker that attempts authentication with the same CredentialRequest twice and receives
```

There is one form of leakage that the above allows and whose prevention would
require a change in OPAQUE.

> Note that an attacker that tests an idU (and same M) twice and receives

Reviewer: ```suggestion
An attacker that attempts authentication with the same CredentialRequest twice and receives
```
